pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/forward/ForwardIndexPlugin.java

```diff
@@ -26,10 +26,8 @@
 @AutoService(IndexPlugin.class)
 public class ForwardIndexPlugin implements IndexPlugin<ForwardIndexType> {
 
-  private static final ForwardIndexType INSTANCE = new ForwardIndexType();
```

Comment:
@gortiz Can you help take a look at this part? What is the correct way to plug-in a custom `ForwardIndexType`? Should it be handled with `@AutoService`?

Reply:
I don't think that is the correct way to do it. Instead, I would recommend either:
- Using the current system, a fork can set a different IndexService instead of using the default. This can easily be done by making IndexService.init() public and then calling IndexService.setInstance()
- Add a priority for each IndexPlugin, so if there is a collision on id, IndexService.fromServiceLoader() only loads the one with higher priority.

Reply:
I've added the concept of priority in https://github.com/apache/pinot/pull/15766. Please take a look